### PR TITLE
 Clean up flags.rb code 

### PR DESCRIPTION
**Pull request by marielfrank**

### CODE REVIEW

File: lib/colorls/flags.rb [hunk 1]
```diff
@@ -23,38 +23,39 @@ def initialize(*args)
 
     def process
       incompatible = report_incompatible_flags
-      return STDERR.puts "\n   #{incompatible}".colorize(:red) if incompatible
+      STDERR.puts "\n   #{incompatible}".colorize(:red) if incompatible
```
Comment (athityakumar) [inline lib/colorls/flags.rb]: I think this `return` needs to stay, to stop the program from going to the below if-else when there are incompatible tags.

File: lib/colorls/flags.rb [hunk 2]
```diff
@@ -23,38 +23,39 @@ def initialize(*args)
 
     def process
       incompatible = report_incompatible_flags
```
Comment (athityakumar) [inline lib/colorls/flags.rb]: Remove variable `incompatible` and rename method `report_incompatible_flags` to `incompatible_flags?`, to be used in this method.

File: lib/colorls/flags.rb [hunk 3]
```diff
@@ -23,38 +23,39 @@ def initialize(*args)
...
-        @colors = ColorLS.load_from_yaml('dark_colors.yaml', true)
+        color_scheme = 'dark_colors.yaml'
         STDERR.puts "\n Restrain from using --light and --dark flags together."
           .colorize(@colors[:error])
       elsif light_colors
-        @colors = ColorLS.load_from_yaml('light_colors.yaml', true)
+        color_scheme = 'light_colors.yaml'
       else # default colors
-        @colors = ColorLS.load_from_yaml('dark_colors.yaml', true)
+        color_scheme = 'dark_colors.yaml'
```
Comment (athityakumar) [inline lib/colorls/flags.rb]: Take the `color_scheme` variable outside the if-else block, like

```ruby
if light_colors && dark_colors
  STDERR.puts "\n Restrain from using --light and --dark flags together."
    .colorize(@colors[:error])
end

color_scheme_file = light_colors ? 'light_colors.yaml' : 'dark_colors.yaml'
@colors = ColorLS.load_from_yaml(color_scheme_file, true)
```

**Comment by marielfrank** [comment]
Working on these changes now! Totally agree that explicit returns are more readable, but they are slower in Ruby so I was trying to make things slightly faster ;-) (https://blog.nerdbucket.com/the-horrible-cost-of-using-explicit-returns-in-ruby/article, http://lizard-spock.co.uk/blog/tech/benchmarking-ruby-explicit-vs-implicit-returns/)

**Comment by marielfrank** [comment]
Please let me know if you need any further changes :)

**Pull request merged**